django/db/backends/sqlite3/features.py

```diff
@@ -70,11 +70,11 @@ def introspected_field_types(self):
 
     @cached_property
     def supports_json_field(self):
-        try:
-            with self.connection.cursor() as cursor, transaction.atomic():
+        with self.connection.cursor() as cursor, transaction.atomic():
+            try:
                 cursor.execute('SELECT JSON(\'{"a": "b"}\')')
-        except OperationalError:
-            return False
+            except OperationalError:
+                return False
         return True
```

Comment:
I think `self.connection.alias` should be used in `transaction.atomic`, exceptions also need to be able to bubble through `transaction.atomic`

```python
@cached_property
def supports_json_field(self):
    with self.connection.cursor() as cursor:
        try:
            with transaction.atomic(self.connection.alias):
                cursor.execute('SELECT JSON(\'{"a": "b"}\')')
        except OperationalError:
            return False
    return True

Reply:
Thanks :+1: Updated.